pkg/diff/diff.go

```diff
@@ -660,13 +667,44 @@ func (sc *Syncer) Solve(ctx context.Context, parallelism int, dry bool, isJSONOu
 				if oldPlugin, ok := e.OldObj.(*state.Plugin); ok {
 					oldPluginCopy := &state.Plugin{Plugin: *oldPlugin.DeepCopy()}
 					e.OldObj = oldPluginCopy
+					linkedPartialConfig, err := utils.FindLinkedPartials(ctx, sc.kongClient, &oldPluginCopy.Plugin)
+					if err != nil {
+						return nil, err
+					}
+
+					err = kong.FillPluginsDefaultsWithPartials(&oldPluginCopy.Plugin, schema, linkedPartialConfig)
+					if err != nil {
+						return nil, fmt.Errorf("failed processing auto fields: %w", err)
+					}
+
 					if err := kong.ClearUnmatchingDeprecations(&pluginCopy.Plugin, &oldPluginCopy.Plugin, schema); err != nil {
 						return nil, fmt.Errorf("failed processing auto fields: %w", err)
 					}
 				}
 			}
 		}
 
+		// If the event is for a partial, inject defaults in the partial's config
+		// that will be used for the diff. This is needed to avoid highlighting
+		// default values that were populated by Kong as differences.
+		if partial, ok := e.Obj.(*state.Partial); ok {
+			partialCopy := &state.Partial{Partial: *partial.DeepCopy()}
+			e.Obj = partialCopy
+
+			if exists, _ := utils.WorkspaceExists(ctx, sc.kongClient); exists {
+				var schema map[string]interface{}
+				schema, err = sc.kongClient.Partials.GetFullSchema(ctx, partialCopy.Partial.Type)
```

Comment:
Any reason to reuse the `err`? AFAICT it's only used in the line below. Line 701 already declares its own `err`.

```suggestion
				schema, err := sc.kongClient.Partials.GetFullSchema(ctx, partialCopy.Partial.Type)
```